src/transformers/models/trillsson_efficientnet/modeling_trillsson_efficientnet.py

```diff
@@ -0,0 +1,688 @@
+        self.gradient_checkpointing = False
+
+    def forward(self, input_features):
+        hidden_states = self.stem_conv(input_features)
+        hidden_states = self.stem_bn(hidden_states)
+        hidden_states = self.activation(hidden_states)
+        return hidden_states
+
+
+class TrillssonTop(nn.Module):
+    def __init__(
+        self,
+        config: TrillssonEfficientNetConfig,
+        in_channels: int,
+        out_channels: int,
+    ) -> None:
+        super().__init__()
+        self.config = config
+        self.top_conv = Conv2DSamePadding(
+            in_channels=in_channels, out_channels=out_channels, kernel_size=1, stride=1, padding=0, bias=False
+        )
+        self.top_bn = nn.BatchNorm2d(
+            num_features=out_channels, eps=self.config.norm_eps, momentum=self.config.norm_momentum
+        )
+        self.activation = ACT2FN[config.hidden_act]
+        self.gradient_checkpointing = False
+
+    def forward(self, input_features):
+        hidden_states = self.top_conv(input_features)
+        hidden_states = self.top_bn(hidden_states)
+        hidden_states = self.activation(hidden_states)
+        return hidden_states
+
+
+class TrillssonInvertedResidual(nn.Module):
+    def __init__(
+        self,
+        config: TrillssonEfficientNetConfig,
+        in_channels: int,
+        out_channels: int,
+        stride: int,
+        expand_ratio: int,
+        use_se: int,
+        survival_probability: int,
+    ) -> None:
+        """TrillssonInvertedResidual block: Inverted Residual Bottleneck."""
+        super().__init__()
+        hidden_dim = round(in_channels * expand_ratio)
+        self.config = config
+        self.expand_ratio = expand_ratio
+        self.use_se = use_se
+        self.survival_probability = survival_probability
+        self.identity = stride == 1 and in_channels == out_channels
+        self.activation = ACT2FN[config.hidden_act]
+        self.dropout = nn.Dropout(self.survival_probability)
+        if self.expand_ratio != 1:
+            self.expand_conv = Conv2DSamePadding(
+                in_channels=in_channels, out_channels=hidden_dim, kernel_size=1, stride=1, padding=0, bias=False
+            )
+            self.expand_bn = nn.BatchNorm2d(
+                num_features=hidden_dim, eps=self.config.norm_eps, momentum=self.config.norm_momentum
+            )
+        self.dwconv2 = Conv2DSamePadding(
+            in_channels=hidden_dim,
+            out_channels=hidden_dim,
+            kernel_size=3,
+            stride=stride,
+            padding=0,
+            groups=hidden_dim,
+            bias=False,
+        )
+        self.bn = nn.BatchNorm2d(num_features=hidden_dim, eps=self.config.norm_eps, momentum=self.config.norm_momentum)
+        if self.use_se:
+            self.se = TrillssonSqueezeExcitation(config=self.config, in_channels=in_channels, out_channels=hidden_dim)
+            self.project_conv = Conv2DSamePadding(
```

Comment:
It is probably a bit too late but let's try if possible to have longer variable names for `bn` and `se`

Reply:
Revise as suggestion